app/models/user.rb

```diff
@@ -120,6 +120,9 @@ class User < ApplicationRecord
 
   scope :random, -> { order('RAND()') }
 
+  scope :donor, -> { where.not(first_donated_at: nil) }
+  scope :supporter, -> { donor.where(show_on_supporters_page: true) }
```

Comment:
Maybe `public_supporter`?

Reply:
(It wasn't clear to me the difference between donor + supporter without reading the code)